src/essence/Tools/Isochrone/IsochroneTool_Manager.js

```diff
@@ -0,0 +1,443 @@
+            max.x += 256;
+        while(isWithinBounds(startPoint.x, max.y))
+            max.y += 256;
+        
+        min = min.divideBy(256);
+        max = max.divideBy(256);
+
+        this.tileBounds = window.L.bounds(min, max);
+
+        this.startPx = Map_.map
+            .project(this.start, this.options.resolution)
+            .subtract(this.tileBounds.min.multiplyBy(256))
+            .floor();
+
+        this.getData();
+    }
+
+    getRequiredTilesList(source) {
+        const {zoomOffset} = source;
+        const zoom = this.options.resolution + zoomOffset;
+        let bounds = this.tileBounds;
+        if(zoomOffset !== 0) { //Handle lower-res tiles
+            const scaleFactor = Math.pow(2, zoomOffset);
+            bounds = window.L.bounds(
+                this.tileBounds.min.multiplyBy(scaleFactor),
+                this.tileBounds.max.multiplyBy(scaleFactor)
+            );
+        }
+        const startPoint = Map_.map.project(this.start, zoom);
+        const startTile = startPoint.clone().divideBy(256).floor();
+        
+        let tileList = [];
+        for(let y = bounds.min.y; y < bounds.max.y; y++) {
+            for(let x = bounds.min.x; x < bounds.max.x; x++) {
+                //Measure distance to start from nearest corner/edge
+                let measureX = startPoint.x;
+                if(x > startTile.x) measureX = x * 256;
+                else if (x < startTile.x) measureX = (x + 1) * 256;
+
+                let measureY = startPoint.y;
+                if(y > startTile.y) measureY = y * 256;
+                else if(y < startTile.y) measureY = (y + 1) * 256;
+
+                const measureLatLng = Map_.map.unproject([measureX, measureY], zoom);
+                const dist = F_.lngLatDistBetween(
+                    this.start.lng,
+                    this.start.lat,
+                    measureLatLng.lng,
+                    measureLatLng.lat
+                );
+
+                if(dist <= this.options.maxRadius) {
+                    tileList.push({
+                        x,
+                        y,
+                        z: zoom,
+                        relX: x - bounds.min.x,
+                        relY: y - bounds.min.y,
+                        dist
+                    });
+                }
+            }
+        }
+
+        tileList.sort((a, b) => a.dist - b.dist);
+        return tileList;
+    }
+
+    getData() {
+        console.time("load");
+        
+        this.queryJobs.map(job => job.stop());
+        this.queryJobs = [];
+
+        let promises = [];
+        for(const dataType of this.modelProto.requiredData) {
+            const sourceIndex = this.options[dataType +"_source"];
+            const source = this.dataSources[dataType][sourceIndex];
+            const requiredTiles = this.getRequiredTilesList(source);
+
+            const job = new QueryJob(source, requiredTiles, this.tileBounds);
+            this.queryJobs.push(job);
```

Comment:
re: your question about prioritizing some data to download first so the user sees something quickly, you could probably break the set of jobs into two lists of promises, one that is short and includes just enough data to render something, then a big list with everything else. then you could do:

`queryShortList.then(Promise.all(allOtherJobs))`

Reply:
my thought there was to load jobs in sequence in the order they came in, so that for instance if a user started Isochrone 2 loading while Isochrone 1 was also loading, Isochrone 2 would wait for Isochrone 1 to finish and display. the tool unfortunately isn't smart enough to start with only part of the data it needs.